synapse/api/auth.py

```diff
@@ -43,30 +42,9 @@
 GUEST_DEVICE_ID = "guest_device"
 
 
-class Auth(object):
-    """
-    FIXME: This class contains a mix of functions for authenticating users
-    of our client-server API and authenticating events added to room graphs.
-    """
-    def __init__(self, hs):
-        self.hs = hs
-        self.clock = hs.get_clock()
-        self.store = hs.get_datastore()
-        self.state = hs.get_state_handler()
-        self.TOKEN_NOT_FOUND_HTTP_STATUS = 401
-
-    @defer.inlineCallbacks
-    def check_from_context(self, event, context, do_sig_check=True):
-        auth_events_ids = yield self.compute_auth_events(
-            event, context.prev_state_ids, for_verification=True,
-        )
-        auth_events = yield self.store.get_events(auth_events_ids)
-        auth_events = {
-            (e.type, e.state_key): e for e in auth_events.values()
-        }
-        self.check(event, auth_events=auth_events, do_sig_check=do_sig_check)
-
-    def check(self, event, auth_events, do_sig_check=True):
+class Auther(object):
+    @staticmethod
+    def check(event, auth_events, do_sig_check=True):
```

Comment:
Would it make sense to make these top level functions rather than staticmethods?